### DISCUSSION TITLE
Display gene products vs gene IDs for a GFF track

### DISCUSSION THREAD

**Discussion by bioinfoMMS** [GMOD/jbrowse-components#2729] [Q&A]
Hi JBrowse2 developers,

Sorry in advance if I have missed this in the user guide, but I was wondering if there is a way to change what label is displayed per gene on a GFF track. I am hoping to modify some of my GFF tracks to show the 'product' from the CDS subfeature for each gene instead of the 'id'. This product value is visible when I click on a gene in the Feature Details panel but I would like it to show up without having to click on a gene. How can I modify the config.json file to do this?

Thanks!

**Comment by garrettjstevens** [comment]
In general, one way to modify a track is to edit it with our graphical editor and then export the config for that track to add to your config file.  On JBrowse Web, make a copy of the track (track menu -> Copy track), which will copy the track to your "Session tracks" (on JBrowse Desktop you don't have to make a copy, you can edit the track directly). Then in the copy, go to the settings (track menu -> Settings). Then you can update what you need, in this case probably the "name" under "labels". You can probably replace "id" with "product" in the existing entry (FYI these types of config slots use [jexl](https://github.com/TomFrost/jexl) to evaluate the expression). Once you're satisfied with the changes, export them (File -> Export session). The resulting JSON file will have a section called "sessionTracks" that contains your modified track that you can then copy into your config.json.

For your change in particular, I would guess the necessary track config would look something like this:

```json
{
  "type": "FeatureTrack",
  "trackId": "gff3_genes",
  "name": "GFF3 genes",
  "assemblyNames": [
    "volvox"
  ],
  "adapter": {
    "type": "Gff3Adapter",
    "gffLocation": {
      "locationType": "UriLocation",
      "uri": "volvox.sort.gff3",
    }
  },
  "displays": [
    {
      "type": "LinearBasicDisplay",
      "displayId": "gff3_genes-LinearBasicDisplay",
      "renderer": {
        "type": "SvgFeatureRenderer",
        "labels": {
          "name": "jexl:get(feature,'product')"
        }
      }
    }
  ]
}
```

**Comment by bioinfoMMS** [comment]
I tried copying and modifying the tracks, but I think its running into a problem because the 'product' is from a sub-feature (CDS) and not a feature. So when I replaced: 

get(feature,'name') || get(feature,'id') 

with 

get(feature,'name') || get(feature,'product')

no labels appeared. Is there a way to specify a label from a sub-feature?

**Reply by garrettjstevens** [reply]
I'm not sure what the structure of your data is, but you can get subfeatures with `get(feature, 'subfeatures')`, so you could do something like this:
```
get(feature, 'subfeatures') != undefined ? get(get(feature, 'subfeatures')[0], 'product') : get(feature, 'name')
```
This only uses the first subfeature, but you might be able to use jexl's [collections filtering](https://github.com/TomFrost/jexl#collections) to access a certain subfeature. If you're interested in writing a plugin, you can write one that uses more complicated functions in JavaScript to determine the label from the feature.

**Reply by bioinfoMMS** [reply]
Since I am working on a bacterial genome, getting the product attribute from first subfeature is fine. Your solution works great. Thank you so much for your help!
